### PR TITLE
fix: Use the QList insert method to perform a validation check, and u…

### PR DESCRIPTION
**Pull request by dengzhongyuan365-dev**
…se append instead of insert when the index value is greater than the size of the list

In Qt6, when the size of the list is exceeded, the insert operation of QList will trigger an assertion crash To avoid the crash, if the size bigger than list's size, use append instead.

The following are included: bookmark, tagdirmenu, workspace ..

log: The insert method of Qlist is compatible.
bug: https://pms.uniontech.com/bug-view-284723.html

### CODE REVIEW

File: src/plugins/filemanager/qml/dfmplugin-workspace/models/fileviewmodel.cpp [hunk 1]
```diff
@@ -614,7 +614,13 @@ QList<ItemRoles> FileViewModel::getColumnRoles() const
         int customCount = roles.count();
         for (auto role : defualtColumnRoleList) {
             if (!roles.contains(role))
-                roles.insert(roles.length() - customCount, role);
+            {
```
Comment (Johnson-zs) [inline src/plugins/filemanager/qml/dfmplugin-workspace/models/fileviewmodel.cpp]: 代码风格似乎没配置

File: src/plugins/filemanager/core/dfmplugin-workspace/utils/filesortworker.cpp [hunk 2]
```diff
@@ -20,6 +20,17 @@ using namespace dfmplugin_workspace;
...
+namespace {
+template<class T>
+void insertToList(QList<T> &list, int index, const T &t)
+{
+    if (index < 0 || index > list.size())
+        list.append(t);
+    else
+        list.insert(index, t);
+}
+}
```
Comment (Johnson-zs) [inline src/plugins/filemanager/core/dfmplugin-workspace/utils/filesortworker.cpp]: namespace后面需要加注释 // namespace XXX

File: src/plugins/filemanager/qml/dfmplugin-workspace/models/fileviewmodel.cpp (hunk 1 above)
Comment (dengzhongyuan365-dev) [reply src/plugins/filemanager/qml/dfmplugin-workspace/models/fileviewmodel.cpp]: 图片
已经使用插件对改动文件进行格式化，但是会出现异常格式化

File: src/plugins/filemanager/core/dfmplugin-workspace/utils/filesortworker.cpp (hunk 2 above)
Comment (dengzhongyuan365-dev) [reply src/plugins/filemanager/core/dfmplugin-workspace/utils/filesortworker.cpp]: 已经修改

**Review by Johnson-zs — APPROVED** [review]

**Pull request merged**